app/assets/javascripts/registrations.js

```diff
@@ -39,10 +39,21 @@ $(document).on('turbolinks:load', function() {
     }
   };
 
-  $("#registration_access_levels").on('change', function() {
+  const hidePaymentInfoIfNeeded = function(value) {
+    const val = parseInt(value);
+    if (window.freeTickets && (window.freeTickets.indexOf(val) !== -1)) {
+      $("#payment-info").hide();
+    } else {
+      $("#payment-info").show();
+    }
+  };
+
+  $("#registration_access_level").on('change', function() {
+    hidePaymentInfoIfNeeded($(this).val());
     return hideCommentFieldIfNeeded($(this).val());
```

Comment:
Return can go here as well, was there beforehand but it doesn't hurt to clean it up a bit